specs.js

```diff
@@ -0,0 +1,149 @@
+var describe = require('mocha').describe
+var it = require('mocha').it
+var expect = require('chai').expect
+var nock = require('nock')
+
+var mockURL = 'http://localhost'
+var mockToken = 'DEADBEEF'
+
+// setup mock for trimperson api
+
+describe('when using trimperson api', function () {
+  var trim
+
+  beforeEach(function () {
+    trim = require('./index')({
+      url: mockURL,
+      token: mockToken,
+      mock: false
+    })
+  })
+
+  it('should create a trimperson object', function () {
+    expect(trim).to.be.a('object')
+  })
+
+  describe('when using the getContainer function', function () {
+    it('should exist on the trim instance', function () {
+      expect(trim.getContainer).to.be.a('function')
+    })
+
+    it('should getContainer and execute the callback', function (done) {
+      var fakeID = 'fakecontainerid'
+
+      var getContainerMock = nock(mockURL)
+        .get('/GetContainer')
+        .query({trimid: fakeID, securityToken: mockToken})
+        .reply(200, {
+          containerNo: 'someid',
+          subContainers: []
+        })
+
+      trim.getContainer(fakeID, function (err, data) {
+        expect(err).to.be.an('null')
+        expect(data).not.to.be.a('undefined')
+
+        expect(data.containerNo).to.equal('someid')
+        expect(data.subContainers).to.be.a('array')
+        getContainerMock.done() // throws an error if no request was recorded
+        done()
+      })
+    })
+  })
+
+  describe('when using the getDocument function', function () {
+    it('should exist on the trim instance', function () {
+      expect(trim.getDocument).to.be.a('function')
+    })
+
+    it('should getDocument succesfully', function (done) {
+      var documentId = 'someIDthatPointsToADoucment'
+      var getDocumentMock = nock(mockURL)
+        .get('/get')
+        .query({id: documentId, securityToken: mockToken})
+        .reply(200, {
+          msg: 'response' // TODO what shape object does getDocument actually return
```

Comment:
getDocument responds with the actual file bytecode, e.g. a PDF file. It's not a json